### DISCUSSION TITLE
How could I use the CLI in other project?

### DISCUSSION THREAD

**Discussion by negahama** [eclipse-langium/langium#1671] [Q&A] [answered]
Hello, everyone
I've made my CLI, So I want to use this CLI in my other project. but I don't know how to do it.
Is the CLI generated to stand-alone executive file by the Langium?
I think my question is so stupid...
Sorry to bother you with such like this question
I am not good at web development. I hope kind and useful response as usual
Thank you!

**Comment by msujew** [comment]
Hey @negahama,

Can you explain a bit how you want to use the CLI in your other project? Is your other project also a JavaScript/TypeScript app?

**Reply by negahama** [reply]
@msujew Yes. What I made is transpiler to transpile my DSL to TypeScript codes.
So I want to append my CLI program to other TypeScript app to convert my DSL codes.. 
I hope my CLI program would be operated like tsc (TypeScript Compiler).

**Reply by msujew** [reply, marked as answer]
In theory, you could just run `node <path/to/cli.js>` to run the transpiler as a separate process. However I would recommend something different:

You can add a dependency from your app to the cli package and invoke the transpiler from code. This has a few benefits, mainly:

* Allows to debug your transpiler (if you need to) from the other app
* Better error handling - usually a CLI just exits with status 1 and tells the user relatively little what went wrong. A programming API could have better access to any internal error of the transpiler.

**Comment by negahama** [comment]
Thank you so much.
Do you happened to know some samples or articles to show me?
There is "A picture is worth a thousand words"

**Reply by msujew** [reply]
We do have an example for something similar available over at the langium-lox repo. See [here for the interpreter](https://github.com/TypeFox/langium-lox/blob/main/langium/src/interpreter/runner.ts) and [here for the notebool kernel that uses that interpreter](https://github.com/TypeFox/langium-lox/blob/main/vscode/src/notebook/lox-notebook-kernel.ts).

Whether its a transpiler or interpreter barely matters, they are very similar from an architectural perspective.
